fix(studio): empty team drop downs

kemaldotearth opened the pull request:
This fixes dropdowns returning empty on team invites when a user is signed in as a developer. For now it adds conditionals to show the option is not available for that role, but perhaps we can handle a little more gracefully.

## I have read the [CONTRIBUTING.md](https://github.com/supabase/supabase/blob/master/CONTRIBUTING.md) file.

YES

## What kind of change does this PR introduce?

This fixes a small UI bug within the Teams area member list, where users signed in with the `Developer` role see empty drop down menus on pending invites.

## What is the current behavior?

Current behaviour returns as follows:
<img width="1021" height="387" alt="Screenshot 2025-07-29 at 13 19 18" src="https://github.com/user-attachments/assets/dbd9a5df-cc2f-49b2-a0aa-b5d3aa646389" />


## What is the new behavior?

For the time being I've updated the conditional checks on `canRevokeInvite` and `canSendInvite` to be `if` statements. This ensures we handle it somewhat gracefully for now. An open question is, can the `Developer` role do anything with the dropdown menu anyway? Might be worth hiding it from their scope?

Updated behaviour is like so:
<img width="861" height="449" alt="Screenshot 2025-07-29 at 14 40 17" src="https://github.com/user-attachments/assets/b31e18a1-dc03-4838-88dc-5ed1289ac63f" />


## Additional context

Please let me know if this is taken care of in another PR, if so, feel free to close this!

Commit:
fix: empty team drop downs

This fixes dropdowns returning empty on team invites when a user is signed in as a developer. For now it adds conditionals to show the option is not available for that role, but perhaps we can handle a little more gracefully.

joshenlim commented:
> i guess that would require changes elsewhere

Yup - whichever way we decide it'll be best to keep it as a consistent pattern across the dashboard. We can do the tooltips for now and revisit this as a whole again separately 🙂

Commit:
fix: change from inline to tooltips on dropdowns

This updates the members table dropdown to use the tooltip as opposed to inline message for disabled items.

joshenlim committed:
Tiny clean up

joshenlim committed:
Biiiit more nit

joshenlim reviewed (approved):
LGTM! Verified changes locally 😄
